tests/integration/querysets_test.py

```diff
@@ -0,0 +1,56 @@
+import datetime as dt
+
+from django.test import TestCase
+
+from nav.models.event import AlertHistory
+from nav.models.event import AlertHistoryVariable
+from nav.models.event import EventType
+from nav.models.event import Subsystem
+from nav.models.fields import INFINITY
+from nav.models.manage import Netbox
+from nav.models.manage import Category
+from nav.models.manage import Room
+from nav.models.manage import Organization
+
+
+class NetboxQuerysetTest(TestCase):
+
+    def setUp(self):
+        # Some rows have already been created
+        _netbox_data = {
+            "room": Room.objects.get(id="myroom"),
+            "organization": Organization.objects.get(id="myorg"),
+            "category": Category.objects.get(id="SW")
+        }
+        TEST_NETBOX_DATA = [
+            dict(sysname="foo.bar.com", ip="158.38.152.169", **_netbox_data),
+            dict(sysname="bar.bar.com", ip="158.38.152.231", **_netbox_data),
+            dict(sysname="spam.bar.com", ip="158.38.152.9", **_netbox_data),
+        ]
+
+
+        self.netboxes = [Netbox.objects.create(**netbox_data)
+                         for netbox_data in TEST_NETBOX_DATA]
+        ah = AlertHistory.objects.create(
+            source=Subsystem.objects.first(),
+            netbox=self.netboxes[2],
+            event_type=EventType.objects.get(id="maintenanceState"),
+            start_time=dt.datetime.now(), value=0,
+            end_time=INFINITY,  # UNRESOLVED
+            severity=0
+        )
+        AlertHistoryVariable.objects.create(alert_history=ah,
+                                            variable="netbox")
+
+
+    def test_on_maintenance_true(self):
+        on_maintenance = Netbox.objects.on_maintenance(True)
+        self.assertEqual(on_maintenance.count(), 1)
+        self.assertEqual(on_maintenance[0], self.netboxes[2])
+
+    def test_on_maintenance_false(self):
+        # A netbox not used in this test has already been created
+        not_on_maintenance = Netbox.objects.on_maintenance(False)
+        self.assertIn(self.netboxes[0], not_on_maintenance)
+        self.assertIn(self.netboxes[1], not_on_maintenance)
+        self.assertNotIn(self.netboxes[2], not_on_maintenance)
```

Comment:
You mentioned you didn't know who to write this as a pure pytest test. Here's how I would have written it (for the learn):

```python
import datetime as dt

import pytest

from nav.models.event import AlertHistory
from nav.models.event import AlertHistoryVariable
from nav.models.event import EventType
from nav.models.event import Subsystem
from nav.models.fields import INFINITY
from nav.models.manage import Netbox
from nav.models.manage import Category
from nav.models.manage import Room
from nav.models.manage import Organization


class TestNetboxQueryset(object):
    def test_on_maintenance_true(self, db, netboxes):
        on_maintenance = Netbox.objects.on_maintenance(True)
        assert on_maintenance.count() == 1
        assert on_maintenance[0] == netboxes[2]

    def test_on_maintenance_false(self, db, netboxes):
        # A netbox not used in this test has already been created
        not_on_maintenance = Netbox.objects.on_maintenance(False)
        assert netboxes[0] in not_on_maintenance
        assert netboxes[1] in not_on_maintenance
        assert netboxes[2] not in not_on_maintenance


# Fixtures


@pytest.fixture
def netboxes(db):
    _netbox_data = {
        "room": Room.objects.get(id="myroom"),
        "organization": Organization.objects.get(id="myorg"),
        "category": Category.objects.get(id="SW"),
    }
    TEST_NETBOX_DATA = [
        dict(sysname="foo.bar.com", ip="158.38.152.169", **_netbox_data),
        dict(sysname="bar.bar.com", ip="158.38.152.231", **_netbox_data),
        dict(sysname="spam.bar.com", ip="158.38.152.9", **_netbox_data),
    ]

    netboxes = [
        Netbox.objects.create(**netbox_data) for netbox_data in TEST_NETBOX_DATA
    ]
    ah = AlertHistory.objects.create(
        source=Subsystem.objects.first(),
        netbox=netboxes[2],
        event_type=EventType.objects.get(id="maintenanceState"),
        start_time=dt.datetime.now(),
        value=0,
        end_time=INFINITY,  # UNRESOLVED
        severity=0,
    )
    AlertHistoryVariable.objects.create(alert_history=ah, variable="netbox")
    return netboxes
```